changelog/8074.improvement.md

```diff
@@ -0,0 +1 @@
+Rasa test now prints a warning if the test stories contain bot utterances that are not part of the domain.
```

Comment:
From a user perspective these are from test stories, not the domain right?

Reply:
*train stories

Reply:
Although I guess from a user perspective that might be confusing as well, since this problem is unique to e2e - we always warned users if their bot utterances were not defined, like this:

`Action 'utter_1' is listed as a response action in the domain file, but there is no matching response defined`

Reply:
Wait, no this really is about the test stories, right? The action_text comes from the training stories, and the present PR is about warning the user when the test stories contain an action_text that did not occur in the training set.

Reply:
Eek, yes, I'm wrong. Sorry about that, I got mixed up.

Reply:
Does this mean we aren't checking whether the `action_names` in test stories are defined?
So if I have a test story like:
```
stories:
- story: ...
  steps:
    ...
    - bot: utter_1
    ...
 ```
 and `utter_1` isn't defined in the domain there's no warning?

Reply:
In your example, you would now get a warning, because `utter_1` is an `action_text` (not an `action_name`). The `action_text`s don't appear in the domain anyway.